src/back/space/src/service/service.cpp

```diff
@@ -0,0 +1,397 @@
+
+#include <regex>
+#include <boost/uuid/uuid_generators.hpp>
+
+#include "../repo/repository.hpp"
+#include "../../../shared/errors.hpp"
+#include "../../../shared/paging.hpp"
+
+#include <userver/yaml_config/merge_schemas.hpp>
+#include "userver/components/component_config.hpp"
+#include "userver/components/component_context.hpp"
+
+namespace svetit::space {
+
+/*static*/ yaml_config::Schema Service::GetStaticConfigSchema()
+{
+	return yaml_config::MergeSchemas<components::LoggableComponentBase>(R"(
+type: object
+description: Main service component
+additionalProperties: false
+properties:
+  can-create:
+    type: string
+    description: Can create new spaces
+  default-space:
+    type: string
+    description: Default space to join
+  spaces-limit-for-user:
+    type: string
+    description: How many spaces user can join
+  items-limit-for-list:
+    type: string
+    description: How many items list may contain
+)");
+}
+
+Service::Service(
+		const components::ComponentConfig& conf,
+		const components::ComponentContext& ctx)
+	: components::LoggableComponentBase{conf, ctx}
+	, _repo{ctx.FindComponent<Repository>()}
+	, _canCreate{conf["can-create"].As<bool>()}
+	, _defaultSpace{conf["default-space"].As<std::string>()}
+	, _spacesLimitForUser{conf["spaces-limit-for-user"].As<int>()}
+	, _itemsLimitForList{conf["items-limit-for-list"].As<int>()}
+{}
+
+bool Service::IsListLimit(int limit) {
+	return _itemsLimitForList < limit;
+}
+
+PagingResult<model::Space> Service::GetList(const std::string& userId, unsigned int start, unsigned int limit)
+{
+	if (!_defaultSpace.empty()) {
+		const auto defSpace = _repo.Space().SelectByKey(_defaultSpace);
+		if (!_repo.SpaceUser().IsUserInside(defSpace.id, userId)) {
+			// todo - what default role must be set here?
+			_repo.SpaceUser().Insert(defSpace.id, userId, false, Role::Type::User);
+		}
+	}
+	return _repo.SelectByUserId(userId, start, limit);
+}
+
+PagingResult<model::Space> Service::GetAvailableList(const std::string& userId, unsigned int start, unsigned int limit)
+{
+	return _repo.SelectAvailable(userId, start, limit);
+}
+
+PagingResult<model::Space> Service::GetAvailableListBySpaceName(const std::string& spaceName, const std::string& userId, unsigned int start, unsigned int limit)
+{
+	return _repo.SelectAvailableBySpaceName(spaceName, userId, start, limit);
+}
+
+PagingResult<model::SpaceInvitation> Service::GetInvitationList(unsigned int start, unsigned int limit, const std::string& userId)
+{
+	return _repo.SelectInvitationsForSpaceList(start, limit, userId);
+}
+
+PagingResult<model::SpaceInvitation> Service::GetInvitationListBySpaceForSpaceDetail(const std::string& spaceId, unsigned int start, unsigned int limit, const std::string& userId)
+{
+	const auto spaceUuid = utils::BoostUuidFromString(spaceId);
+	if (_repo.SpaceUser().IsAdmin(spaceUuid, userId))
+		return _repo.SpaceInvitation().SelectBySpace(spaceUuid, start, limit);
+	else
+		throw errors::Unauthorized();
```

Comment:
Если оставить без явного return то компилятор может начать сыпать варнинги. Давай перепишем так.

```suggestion
	if (!_repo.SpaceUser().IsAdmin(spaceUuid, userId))
		throw errors::Unauthorized();
	return _repo.SpaceInvitation().SelectBySpace(spaceUuid, start, limit);
```